### DISCUSSION TITLE
Cannot update entity with generated column

### DISCUSSION THREAD

**Discussion by kareldonk** [mikro-orm/mikro-orm#5660] [General]
### Describe the bug

I have an entity with a generated column in PostgreSQL. Defined similar to:

```
export class TestEntity extends EntityBase {
    @PrimaryKey({ autoincrement: true, type: 'bigint' })
    public id: string;

    @Property({ nullable: false, type: 'text' })
    public content: string;

    @Property({ generated: `(to_tsvector('english', content)) stored`, nullable: false, type: 'tsvector' })
    @Index({
        expression: 'create index "full_text_search_idx" on "test_entity" using gin(full_text_search)',
        name: 'full_text_search_idx',
    })
    public fullTextSearch: Opt & string;
}
```

I can start a transaction and create a new entity and save it using the `insert` function from the entitymanager. However, when I retrieve the entity with the `find` function right after that, and then commit the transaction I get an error similar to this:
```
DriverException: update "test_entity" set "full_text_search" = to_tsvector('word') where "id" = '100' returning "full_text_search" - column "full_text_search" can only be updated to DEFAULT
 - detail: Column "full_text_search" is a generated column.
```

It seems like mikro-orm is detecting a change in the generated column after retrieving it with the `find` function and tries to update it again with an update SQL query, which fails.
Shouldn't mikro-orm detect which columns are generated and ignore those for updates/inserts?

### Reproduction

- Define entity with generated column
- Begin transaction
- Insert entity
- Retrieve inserted entity
- Commit transaction

### What driver are you using?

@mikro-orm/postgresql

### MikroORM version

6.2.7

### Node.js version

20 LTS

### Operating system

Windows 10

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Slack](https://join.slack.com/t/mikroorm/shared_invite/enQtNTM1ODYzMzM4MDk3LWM4ZDExMjU5ZDhmNjA2MmM3MWMwZmExNjhhNDdiYTMwNWM0MGY5ZTE3ZjkyZTMzOWExNDgyYmMzNDE1NDI5NjA).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
I am not able to reproduce this, please provide a complete repro. Maybe you are setting the value yourself? (and why?) It's a generated column, you should let the db provide it, that could indeed cause this.

**Reply by B4nan** [reply]
Just tried, and you are correct that we don't ignore those columns when computing changesets - but I am not entirely sure if we should ignore them or use `default` as the error suggests instead, since the value is stored and you want to update it?

edit: probably not, the value is correct after updating the actual columns without that

**Reply by kareldonk** [reply]
Yes so when the entity is stored, and then retrieved, the generated column value is updated with what the database generated. This is good because we do want to read the values of generated columns when getting an entity. The problem is that we should not save changes to the value back to the DB. In my case, I was not changing the value of the column myself (I tripple checked this), but somehow mikro-orm detected that the value was changed and it tries to update the column when committing the transaction. I'm not sure if you tested with the exact same `tsvector` type and generated definition like I put in my example above, but the change detection getting triggered could be related to that specific type and definition.

**Comment by B4nan** [comment]
This should help 55dfbf900911f32068096e412191d2d25415ed80

**Reply by kareldonk** [reply]
Thanks, is this update in a specific version right now, or do i have to manually patch for now?

**Reply by B4nan** [reply]
You can use the latest dev version, every commit to master is published.

**Reply by kareldonk** [reply]
Any ETA on when this fix will be published in non-dev release?

**Reply by B4nan** [reply]
hopefully during this week, v6.3 is almost ready

**Reply by kareldonk** [reply]
Thank you
